File: Problem Set1/Cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    //Ask the user for amount of change (only positive)
    float change;
    do
    {
        change = get_float("Change owed: ");
    }
    while (change < 0);
    // change dollars for cents
    int cents = round(change * 100);
    // define the coins that can be used
    int penys = 1;
    int nickels = 5;
    int dimes = 10;
    int quarters = 25;

    int rounds = 0;
    // number of quarters
    while (cents >= 25)
    {
        cents = cents - quarters;
        rounds++;
    }
    // number of dimes
    while (cents >= 10)
    {
        cents = cents - dimes;
        rounds++;
    }
    // number of nickels
    while (cents >= 5)
    {
        cents = cents - nickels;
        rounds++;
    }
    // number of penys
    while (cents > 0)
    {
        cents = cents - penys;
        rounds++;
    }

    printf("%i \n", rounds);
}

File: Problem Set6/cash/cash.py
from cs50 import get_float


def get_input():
    change = get_float("Change owed: ")
    return change


def checking_input(change):
    if change > 0:
        change = round(change * 100)
        counts = counting(change)
        return counts
    else:
        get_input()


def counting(change):
    counts = 0
    while change > 0:
        if change >= 25:
            change -= 25
            counts += 1
        elif change < 25 and change >= 10:
            change -= 10
            counts += 1
        elif change < 10 and change >= 5:
            change -= 5
            counts += 1
        elif change < 5 and change >= 1:
            change -= 1
            counts += 1
    return counts


def main():

    change = get_input()
    counts = checking_input(change)
    print(counts)


main()
